scopesim/server/database.py

```diff
@@ -7,141 +7,193 @@
+        Other options are:
+        - "stable" : the most recent stable version
+        - "latest" : the latest development version to be published
+        - a specific package filename as given by list_packages (see examples)
+        - a github url for the specific branch and package (see examples)
+
+    save_dir : str, optional
+        The place on the local disk where the ``.zip`` package is to be saved.
+        If left as None, defaults to the value in
         scopesim.rc.__config__["!SIM.file.local_packages_path"]
 
-    return_pkgs : bool
-        If True, returns a list of package names
-
-    silent : bool
-        If True, does not print the list of package names
+    from_cache : bool, optional
+        Use the cached versions of the packages. If None, defaults to the RC
+        value: ``!SIM.file.use_cached_downloads``
 
     Returns
     -------
-    all_pkgs : list of str
-        A list of paths to the ``.zip`` packages relative to ``url``
-        The full string should be passed to download_package
+    save_path : str
+        The absolute path to the saved ``.zip`` package
 
-    """
+    Examples
+    --------
+    ::
+        from scopesim import download_packages, list_packages
 
-    def print_package_list(the_pkgs, loc=""):
-        print(f"\nPackages saved {loc}\n" + "=" * (len(loc) + 15))
-        for pkg in the_pkgs:
-            print(pkg)
+        # Stable release of a list of packages
+        download_packages(["test_package", "test_package"])
 
-    if url is None:
-        url = rc.__config__["!SIM.file.server_base_url"]
-    if local_dir is None:
-        local_dir = rc.__config__["!SIM.file.local_packages_path"]
-
-    return_pkgs_list = []
-
-    if location.lower() in ["local", "all"]:
-        local_pkgs = get_local_packages(local_dir)
-        if not silent:
-            print_package_list(local_pkgs, f"locally: {local_dir}")
-            return_pkgs_list += local_pkgs
-
-    if location.lower() in ["server", "all"]:
-        server_pkgs = []
-        folders = get_server_elements(url, "/")
-        for folder in folders:
-            pkgs = get_server_elements(url + folder, ".zip")
-            server_pkgs += [folder + pkg for pkg in pkgs]
-        if not silent:
-            print_package_list(server_pkgs, f"on the server: {url}")
-            return_pkgs_list += server_pkgs
-
-    if return_pkgs:
-        return return_pkgs_list
+        # Development release of a single package
+        download_packages("test_package", release="latest")
 
-    return None
+        # Specific version of the package
+        list_packages("test_package")
+        download_packages("test_package", release="2022-04-09.dev")
+
+        # Specific package from a Gtihub commit hash (use "@" or ":")
+        download_packages("ELT", release="github:728761fc76adb548696205139e4e9a4260401dfc")
+        download_packages("ELT", release="github@728761fc76adb548696205139e4e9a4260401dfc")
```

Comment:
It seems that any git label can be used, e.g. also tags and branches:
```suggestion
        # Specific package from a Gtihub commit hash or branch/tag name (use "@" or ":")
        download_packages("ELT", release="github:728761fc76adb548696205139e4e9a4260401dfc")
        download_packages("ELT", release="github@728761fc76adb548696205139e4e9a4260401dfc")
        download_packages("ELT", release="github@dev_master")
```